File: android/feature/rewards/src/main/java/com/enterprise/feature/rewards/domain/entities/RewardsState.kt
package com.enterprise.feature.rewards.domain.entities

import com.enterprise.core.domain.entities.DomainError
import com.enterprise.core.domain.entities.Reward
import com.enterprise.core.domain.entities.RewardsAccount

/** Rewards state for MVI pattern. */
data class RewardsState(
    val rewards: List<Reward> = emptyList(),
    val account: RewardsAccount? = null,
    val isLoading: Boolean = false,
    val error: DomainError? = null,
    val selectedReward: Reward? = null
)

File: ios/EnterpriseMobileApp/Sources/Feature/Rewards/Domain/Entities/RewardsState.swift
import Foundation
import CoreDomain

/// Rewards feature state.
public struct RewardsState: Equatable, Sendable {
    public var rewards: [Reward]
    public var account: RewardsAccount?
    public var isLoading: Bool
    public var error: DomainError?
    public var selectedReward: Reward?

    public init(
        rewards: [Reward] = [],
        account: RewardsAccount? = nil,
        isLoading: Bool = false,
        error: DomainError? = nil,
        selectedReward: Reward? = nil
    ) {
        self.rewards = rewards
        self.account = account
        self.isLoading = isLoading
        self.error = error
        self.selectedReward = selectedReward
    }
}
